storage/store.go

```diff
@@ -130,8 +135,38 @@ func (s *Store) Close() error {
 	return s.db.Close()
 }
 
-func defaultConnector(datasource string) (*sql.DB, error) {
-	db, err := sql.Open("clickhouse", datasource)
+func defaultConnector(cfg Configuration) (*sql.DB, error) {
```

Comment:
I don't like having `defaultConnector` and `tlsConnector` especially when both accept full Configuration object. The caller has to know which one to call based on the information in the Configuration. 

```
func createDB(cfg Configuration) (*sql.DB, error) {
  connectionParams := fmt.Sprintf("%s?database=%s&username=%s&password=%s", cfg.Address, cfg.Database, cfg.Username, cfg.Password	)

  if cf.TLS {
    // add secure to connectionParams and load the certificate.
  }
```